src/Simple.OData.Client.Core/Adapter/RequestWriterBase.cs

```diff
@@ -37,7 +37,7 @@ public async Task<ODataRequest> CreateGetRequestAsync(string commandText, bool s
                 ReturnsScalarResult = scalarResult,
                 ResultRequired = true,
             };
-            AssignHeaders(request);
+            
```

Comment:
This could prevent headers being applied to GET requests in the future. I think this would be better to move the logic to RequestWriter AcceptHeaders implementations (V3/V4) to skip adding the Prefer header on GET requests.

Reply:
I can't do more right now, I'm under Linux and the project is not building cause I'm using sdk v3.1.301, Simple.Odata.Client seeems to need v4.6.1 Any idea to help me build your library with this my sdk version ?

Reply:
I've not built this under .NET Core 3.1, but I certainly could make the change.

Reply:
The V3 spec does not explicitly disallow this header apparently: https://www.odata.org/documentation/odata-version-3-0/odata-version-3-0-core-protocol/

Reply:
@watfordgnf I agree, in the v3 disallow of this header is not clear as v4 is for get request, feel free to push the change in my branch. I'd be glad to see this issue fixed :)

Reply:
I wasn't able to push to your branch, but check out #751.

Reply:
Ok, I close this issue.